### PR TITLE
doc(pubsub): implement IAM examples

### LINKED ISSUE #7197: Add samples for IAM policies in Pub/Sub
Now that we have a GA IAM library 🎉  we can write code samples for setting, getting, and testing IAM policies for Cloud Pub/Sub.

The region tags are:

* pubsub_get_subscription_policy [python](https://github.com/googleapis/python-pubsub/blob/f4281c13ea5cd3105f304f930f7394302bf9ba31/samples/snippets/iam.py#L47)
* pubsub_set_subscription_policy [python](https://github.com/googleapis/python-pubsub/blob/f4281c13ea5cd3105f304f930f7394302bf9ba31/samples/snippets/iam.py#L98)
* pubsub_test_subscription_permissions [python](https://github.com/googleapis/python-pubsub/blob/f4281c13ea5cd3105f304f930f7394302bf9ba31/samples/snippets/iam.py#L129)
* pubsub_get_topic_policy [python](https://github.com/googleapis/python-pubsub/blob/f4281c13ea5cd3105f304f930f7394302bf9ba31/samples/snippets/iam.py#L27)
* pubsub_set_topic_policy [python](https://github.com/googleapis/python-pubsub/blob/f4281c13ea5cd3105f304f930f7394302bf9ba31/samples/snippets/iam.py#L69)
* pubsub_test_topic_permissions [python](https://github.com/googleapis/python-pubsub/blob/f4281c13ea5cd3105f304f930f7394302bf9ba31/samples/snippets/iam.py#L153)

The only tricky bit would be to figure out if these compile when IAM is disabled.

### PR DESCRIPTION
**Pull request by coryan**
Fixes #7197

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8917)

### CODE REVIEW

File: google/cloud/pubsub/samples/iam_samples.cc [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+  [](std::string project_id, std::string subscription_id) {
+    auto const subscription =
+        pubsub::Subscription(std::move(project_id), std::move(subscription_id));
+    auto client = iam::IAMPolicyClient(
+        iam::MakeIAMPolicyConnection(pubsub::IAMPolicyOptions()));
+    google::iam::v1::TestIamPermissionsRequest request;
+    request.set_resource(subscription.FullName());
+    request.add_permissions("pubsub.subscriptions.consume");
+    request.add_permissions("pubsub.subscriptions.update");
+    std::cout << "request=" << request.DebugString() << "\n";
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/iam_samples.cc:187]: I don't think we want this print statement

**Review by dbolduc — CHANGES_REQUESTED** [review]

File: google/cloud/pubsub/samples/iam_samples.cc [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+    auto response = client.TestIamPermissions(request);
+    if (!response) throw std::runtime_error(response.status().message());
+    std::cout << "Allowed permissions for subscription "
+              << subscription.FullName() << ":";
+    for (auto const& permission : response->permissions()) {
+      std::cout << " " << permission;
+    }
+    std::cout << "\n";
+  }
+  // [END pubsub_test_topic_permissions]
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/iam_samples.cc:198]: also, the bot is mad about this one. s/topic/subscription/

Comment (coryan) [reply google/cloud/pubsub/samples/iam_samples.cc]: Fixed

File: google/cloud/pubsub/samples/iam_samples.cc (hunk 1 above)
Comment (coryan) [reply google/cloud/pubsub/samples/iam_samples.cc]: Removed

File: google/cloud/pubsub/options.h [hunk 3]
```diff
@@ -284,6 +284,20 @@ using SubscriberOptionList =
...
+ *
+ * To manage the IAM policies of Pub/Sub resources you need to configure the
+ * `google::cloud::IAMPolicyClient` to use `pubsub.googleapis.com` as the
+ * `google::cloud::EndpointOption` and `google::cloud::AuthorityOption`.
+ *
+ * This function returns an object that is initialized with these values, you
+ * can provide additional configuration, or override some of the values before
+ * passing the object to `google::cloud::iam::MakeIAMPolicyConnection`.
+ */
+Options IAMPolicyOptions();
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/pubsub/options.h]: Done

File: google/cloud/pubsub/ci/run_integration_tests_emulator_bazel.sh [hunk 4]
```diff
@@ -32,11 +32,12 @@ bazel_test_args=("$@")
 
 # Configure run_emulators_utils.sh to find the instance admin emulator.
```
[the remark this answers is not in the document]
Comment (coryan) [reply google/cloud/pubsub/ci/run_integration_tests_emulator_bazel.sh]: Fixed.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**